lib/srv/mcp/demo.go

```diff
@@ -0,0 +1,299 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+	"github.com/mark3labs/mcp-go/mcp"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils/mcputils"
+)
+
+const (
+	// DemoServerName is the name of the "Teleport Demo" MCP server.
+	DemoServerName = "teleport-demo"
+
+	// demoServerLabel is a special label used to identity the demo server.
+	demoServerLabel = types.TeleportNamespace + "/mcp-demo-server"
+)
+
+// NewDemoServerApp returns the app definition for the "Teleport Demo" MCP
+// server.
+//
+// The purpose of the "Teleport Demo" MCP server is to provide a quick demo on
+// MCP access without the need for external environment setup on MCP
+// servers. This MCP server is in-memory only and uses stdio transport. Access
+// to this MCP server is the same as any other MCP server (`tsh`, RBAC, audit
+// events, etc.).
+func NewDemoServerApp() (types.Application, error) {
+	app, err := types.NewAppV3(types.Metadata{
+		Name:        DemoServerName,
+		Labels:      map[string]string{demoServerLabel: "true"},
+		Description: "Teleport MCP access demo server",
+	}, types.AppSpecV3{
+		MCP: &types.MCP{
+			Command:       "teleport",
```

Comment:
Why is command "teleport" in this case? If it's just so it's non-empty due to validation, can we update validation instead so it can be empty for the demo server?